Merge new simulation feature into main branch

mrwunderbar666 opened the pull request:
I think the new feature is ready for the main branch. The 1000 simulation runs have worked fine so far.

mrwunderbar666 committed:
add function for simulation

mrwunderbar666 committed:
delete accidentally comitted file

mrwunderbar666 committed:
change name from "delta" to "time_delta"

nicolarighetti committed:
Update simulate_data.R

The function now includes a parameter dataset_size that allows the user to specify the size of the final dataset. Additionally, there is a parameter prop_coord_size to control the proportion of coordinated content relative to the total dataset_size. The function automatically estimates the lambda parameters for both coordinated and non-coordinated Poisson distributions and uses an iterative process to achieve the desired dataset size.

nicolarighetti committed:
Revert "Update simulate_data.R"

This reverts commit 7838cc7302b128e8f4d45228603ddc6f66a11731.

nicolarighetti committed:
Update simulate_data.R

A parameter approx_size has been added, allowing control over the approximate size of the dataset by setting the lambdas appropriately, considering the number of requested accounts. The user can also directly specify lambdas for coordinated and non-coordinated networks, which disables the approx_size parameter and returns a warning.

mrwunderbar666 committed:
validate input parameters

nicolarighetti reviewed (approved):
The documentation needs to be edited a little. I will take care of this.

nicolarighetti committed:
documentation updated

The documentation for the function has been updated to include the new parameters and their description
